src/node/internal/internal_errors.ts

```diff
@@ -703,6 +703,23 @@ export class ERR_CRYPTO_SIGN_KEY_REQUIRED extends NodeError {
   }
 }
 
+export class ERR_TLS_HANDSHAKE_TIMEOUT extends NodeError {
+  constructor() {
+    super('ERR_TLS_HANDSHAKE_TIMEOUT', 'TLS handshake timeout');
+  }
+}
+
+export class ConnResetException extends NodeError {
+  path?: string | undefined;
```

Comment:
For learning purposes: given that the parameter is already optional (`?`), what does adding `| undefined` change in TS?

Reply:
?: means the parameter can be "not passed" the the function. Example: fn() will work "hello?: string". 

On the other hand if you want to pass undefined as the value hello?: string | undefined handles the case of passing undefined as an argument. example: fn(undefined). 

This doesn't really matter for js because if you don't pass a variable it will be undefined by default but if you're dealing with types that uses "| undefined" syntax, this forces you to be explicit about this too. (Shoutout to types/node package maintainers)